apps/lottie-viewer/src/hooks/useJsonFieldsState.ts

```diff
@@ -0,0 +1,39 @@
+import { JsonField } from '@src/configUtils';
+import { useRef, useState } from 'react';
+
+type JsonFieldUpdate = Partial<Pick<JsonField, 'isEnabled'>>;
+
+export function useJsonFieldsState() {
```

Comment:
There was a lot of state/ref logic related to managing the json fields - initializing them, updating them, etc. and I couldn't just have local state to handle rendering them as items in the dropdown/selected items. I also needed to keep track of them for ultimately making the updates on save. The way onConfigure works, the state winds up getting cleared when saving so the ref is for keeping track of the changes so we can batch them and send them on save.